tgui/packages/tgui/interfaces/SmartVend.tsx

```diff
@@ -36,46 +45,57 @@ export const SmartVend = (props) => {
+            <NoticeBox>{data.name} is empty.</NoticeBox>
           ) : (
-            <Table>
-              <Table.Row header>
-                <Table.Cell>Item</Table.Cell>
-                <Table.Cell collapsing />
-                <Table.Cell collapsing textAlign="center">
-                  {data.isdryer ? 'Take' : 'Dispense'}
-                </Table.Cell>
-              </Table.Row>
-              {Object.values(contents).map((value, key) => (
-                <Table.Row key={key}>
-                  <Table.Cell>{value.name}</Table.Cell>
-                  <Table.Cell collapsing textAlign="right">
-                    {value.amount}
-                  </Table.Cell>
-                  <Table.Cell collapsing>
-                    <Button
-                      content="One"
-                      disabled={value.amount < 1}
-                      onClick={() =>
-                        act('Release', {
-                          name: value.name,
-                          amount: 1,
-                        })
-                      }
-                    />
-                    <Button
-                      content="Many"
-                      disabled={value.amount <= 1}
-                      onClick={() =>
-                        act('Release', {
-                          name: value.name,
-                        })
-                      }
-                    />
-                  </Table.Cell>
-                </Table.Row>
-              ))}
-            </Table>
+            Object.keys(contents).map((key) => (
+              <Button
+                color="transparent"
+                key={key}
+                m={1}
+                p={0}
+                height="64px"
+                width="64px"
+                tooltip={contents[key].name}
+                tooltipPosition="bottom"
+                textAlign="right"
+                disabled={contents[key].amount < 1}
+                onClick={() =>
+                  act('Release', {
+                    key: key,
+                    amount: 1,
+                  })
+                }
+              >
+                <DmIcon
+                  fallback={fallback}
+                  icon={contents[key].icon}
+                  icon_state={contents[key].icon_state}
+                  height="64px"
+                  width="64px"
+                />
+                {contents[key].amount > 1 && (
+                  <Button
+                    minWidth="24px"
+                    height="24px"
+                    lineHeight="24px"
+                    textAlign="center"
+                    position="absolute"
+                    left="0"
+                    bottom="0"
+                    fontWeight="bold"
+                    fontSize="14px"
+                    onClick={(e) => {
+                      act('Release', {
+                        key: key,
+                      });
+                      e.stopPropagation();
```

Comment:
this should count as part of the parent button, can this just be made into a box so that it's not a nested button and this isn't needed?

Reply:
These buttons have different purposes, what do you mean? The parent button just releases an item, and the children prompts a text input for the desired amount.